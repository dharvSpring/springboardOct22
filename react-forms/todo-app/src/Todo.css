.Todo {
    margin: 1rem;
}

.Todo > button {
    margin-left: 0.2rem;
    padding: 0.1rem;
}