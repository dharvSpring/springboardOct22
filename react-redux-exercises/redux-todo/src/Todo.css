.Todo > button {
    margin-left: 0.2rem;
}

.Todo {
    margin: 0.5rem;
}