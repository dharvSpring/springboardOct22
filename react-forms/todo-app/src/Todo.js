import React from "react";
import './Todo.css';

function Todo({id, text, handleRemove}) {
    const removeMe = (evt) => {
        handleRemove(id);
    }
    return (
        <div className="Todo">
            {text} 
            <button onClick={removeMe}>X</button>
        </div>
    )
}

export default Todo;