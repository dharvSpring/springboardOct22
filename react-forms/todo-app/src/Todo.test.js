import React from "react";
import { render, fireEvent } from "@testing-library/react";

import Todo from "./Todo";

it("smoketest", function() {
    render(<Todo />);
});

it("snapshot", function() {
    const { asFragment } = render(<Todo />);
    expect(asFragment()).toMatchSnapshot();
});

it("calls remove on click", function () {
    const removeFnMock = jest.fn();
    const { getByText } = render(<Todo id="1" text="Remove Me" handleRemove={removeFnMock} />);
    
    const removeButton = getByText("X");
    fireEvent.click(removeButton);
    expect(removeFnMock).toHaveBeenCalled();
});