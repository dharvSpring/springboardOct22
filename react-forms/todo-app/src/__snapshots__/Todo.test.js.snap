// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
